### PR TITLE
[BIOMAGE-2253] Subset experiment flag

### PR DESCRIPTION
**Pull request by saracastel**
# Description
Adds a new flag called `subset_experiment` that is used in the `construct_qc_config` function. 
It is used to control the QC filters in the subset experiments which must, by default, be all disabled.

# Details
#### URL to issue
N/A

#### Link to staging deployment URL (or set N/A)
N/A

#### Integration test branch
master

# Merge checklist
Your changes will be ready for merging after all of the steps below have been completed.

### Code updates
Have best practices and ongoing refactors being observed in this PR
- [ ] Migrated any selector / reducer used to the new format.

### Manual/unit testing
- [ ] Tested changes using InfraMock locally **or** no tests required for change, e.g. Kubernetes chart updates.
- [ ] Validated that current unit tests for code work as expected and are sufficient for code coverage **or** no unit tests required for change, e.g. documentation update.
- [ ] Unit tests written **or** no unit tests required for change, e.g. documentation update.

### Integration testing
**You must check the box below** to run integration tests on the latest commit on your PR branch.
Integration tests have to pass before the PR can be merged. Without checking the box, your PR
**will not pass** the required status checks for merging.

- [ ] Started end-to-end tests on the latest commit.

### Documentation updates
- [ ] Relevant Github READMEs updated **or** no GitHub README updates required.
- [ ] Relevant Wiki pages created/updated **or** no Wiki updates required.

### Optional
- [ ] Staging environment is unstaged before merging.
- [ ] Photo of a cute animal attached to this PR.

### CODE REVIEW

File: pipeline-runner/R/gem2s-6-construct_qc_config.R [hunk 1]
```diff
@@ -19,6 +21,9 @@ construct_qc_config <- function(scdata_list, any_filtered) {
     auto = TRUE,
     filterSettings = list(FDR = 0.01)
   )
+  if (subset_experiment == T) {
+    classifier_config_to_duplicate$enabled <- FALSE
+  }
```
Comment (gerbeldo) [inline pipeline-runner/R/gem2s-6-construct_qc_config.R:26]: Following the idea from below, couldn't this be added as a logical operation in the `classifier_config_to_duplicate` list? like `!any_filtered && !subset_config` (off the top of my head, did not think about all possibilities)

File: pipeline-runner/R/gem2s-6-construct_qc_config.R [hunk 2]
```diff
@@ -34,7 +39,7 @@ construct_qc_config <- function(scdata_list, any_filtered) {
 
   # mito
   default_mitochondrialContent_config <- list(
-    enabled = TRUE,
+    enabled = !subset_experiment,
```
Comment (gerbeldo) [inline pipeline-runner/R/gem2s-6-construct_qc_config.R:42]: noice

File: pipeline-runner/R/gem2s-6-construct_qc_config.R [hunk 3]
```diff
@@ -140,7 +145,7 @@ get_cellsize_config <- function(scdata_list, config) {
 
 get_sample_mitochondrial_config <- function(scdata_list.sample, config) {
   config.sample <- list(
-    enabled = TRUE,
+    enabled = config$enabled,
```
Comment (gerbeldo) [inline pipeline-runner/R/gem2s-6-construct_qc_config.R:148]: this is a good catch, we never set it as disabled when created, but here we re-set it to TRUE by hardcoding.

File: pipeline-runner/R/gem2s-6-construct_qc_config.R (hunk 1 above)
Comment (gerbeldo) [reply pipeline-runner/R/gem2s-6-construct_qc_config.R:26]: also, TRUE not T (I like the single letter but people complain)

File: pipeline-runner/tests/testthat/test-gem2s-6-construct_qc_config.R [hunk 4]
```diff
@@ -26,7 +26,7 @@ mock_scdata_list <- function() {
 
 test_that("cellsize filter is disabled by default and classifier is pre-filtered", {
   scdata_list <- mock_scdata_list()
-  qc_config <- construct_qc_config(scdata_list, any_filtered = TRUE)
+  qc_config <- construct_qc_config(scdata_list, any_filtered = TRUE, subset_experiment = FALSE)
```
Comment (gerbeldo) [inline pipeline-runner/tests/testthat/test-gem2s-6-construct_qc_config.R:29]: I would add a test with subset_experiment = TRUE, and test that all the important parameters are changed to FALSE. You could even implement it as a snapshot test, might be simpler.

**Review by gerbeldo — COMMENTED** [review]
A couple small comments, other than that looks great.

**Review by kafkasl — COMMENTED** [review]
The PR looks good. However, in my opinion the choice of `subset_experiment` variable name should be improved. My reasoning is as follows:
* reading subset_experiment is a bit ambiguous, a newcomer might wonder if:
* is a declaration? meaning that if that parameter is set, we need to perform the subsetting at some point
* is it information? this is a subsetted experiment?

Instead I suggest naming the variable for what it really does, which is disable all the filters. This is a functionality that might be reused in the future (or might not) but it's easier to read and might help with reuse because it clearly indicates what it does. E.g. `qc_filter_disabled` (I'm fine with other names too)

File: pipeline-runner/R/gem2s-6-construct_qc_config.R (hunk 1 above)
Comment (saracastel) [reply pipeline-runner/R/gem2s-6-construct_qc_config.R]: Good point, `!any_filtered && !subset_experiment` works, I checked all the possibilities. Changed

File: pipeline-runner/tests/testthat/test-gem2s-6-construct_qc_config.R (hunk 4 above)
Comment (saracastel) [reply pipeline-runner/tests/testthat/test-gem2s-6-construct_qc_config.R:29]: Right, I added two simple tests for subset_experiment = TRUE.

**Review by gerbeldo — APPROVED** [review]

**Pull request merged**